hrqb/tasks/employee_leave.py

```diff
@@ -0,0 +1,129 @@
+"""hrqb.tasks.employee_leaves"""
+
+# ruff: noqa: S324
+
+import hashlib
+
+import luigi  # type: ignore[import-untyped]
+import pandas as pd
+
+from hrqb.base.task import (
+    PandasPickleTask,
+    QuickbaseUpsertTask,
+    SQLQueryExtractTask,
+)
+from hrqb.utils import convert_oracle_bools_to_qb_bools, normalize_dataframe_dates
+
+
+class ExtractDWEmployeeLeave(SQLQueryExtractTask):
+    """Query Data Warehouse for employee leave data."""
+
+    stage = luigi.Parameter("Extract")
+
+    @property
+    def sql_file(self) -> str:
+        return "hrqb/tasks/sql/employee_leave.sql"
+
+
+class TransformEmployeeLeave(PandasPickleTask):
+
+    stage = luigi.Parameter("Transform")
+
+    def requires(self) -> list[luigi.Task]:
+        from hrqb.tasks.shared import ExtractQBEmployeeAppointments
+
+        return [
+            ExtractQBEmployeeAppointments(pipeline=self.pipeline),
+            ExtractDWEmployeeLeave(pipeline=self.pipeline),
+        ]
+
+    def get_dataframe(self) -> pd.DataFrame:
+        dw_leaves_df = self.named_inputs["ExtractDWEmployeeLeave"].read()
+        qb_emp_appts_df = self.named_inputs["ExtractQBEmployeeAppointments"].read()
+
+        qb_emp_appts_df = qb_emp_appts_df[["HR Appointment Key", "Record ID#"]].rename(
+            columns={
+                "HR Appointment Key": "hr_appt_key",
+                "Record ID#": "related_employee_appointment_id",
+            }
+        )
+        leaves_df = dw_leaves_df.merge(qb_emp_appts_df, how="left", on="hr_appt_key")
+
+        leaves_df = normalize_dataframe_dates(
+            leaves_df,
+            ["appt_begin_date", "appt_end_date", "absence_date"],
+        )
+
+        # WIP TODO: determine what data points from combination employee leave and
+        # employee appointments determine the field "Accrue Seniority".  For now,
+        # placeholder of blanket "Y" (true) until this is determined.
+        leaves_df["accrue_seniority"] = "Y"
+
+        leaves_df = convert_oracle_bools_to_qb_bools(
+            leaves_df, columns=["paid_leave", "accrue_seniority"]
+        )
+
+        # mint a unique MD5 hash based on leave data
```

Comment:
Not questioning the usage of an MD5 hash but just curious about why since it's not readily apparent

Reply:
Good callout!  Thanks.  I've added some docstring explanations for why an MD5 hash is needed (or some kind of hash) to provide a unique key for `Employee Leave` records.

Once things settle down, and HR is happy with general structure of things, I do think another documentation pass is in order.  That said, I think these tasks can and should serve as more granular documentation for how and why they write particular values.  So even if this is explained more generally somewhere about merge fields, I think this will be helpful for digging in and discovering why this field, for this table, is such as it is.

Reply:
Probably saw already -- and it doesn't help explain it, just a bit how it works -- these test it: https://github.com/MITLibraries/hrqb-client/pull/51/files#diff-fc33a6395ec39699f5a3ab10a0604289f177b2f837c1b92391f3966f44b4bb0dR29-R43